cmd/erasure-object.go

```diff
@@ -1071,32 +1075,27 @@ func (er erasureObjects) PutObjectTags(ctx context.Context, bucket, object strin
 	if err != nil {
 		return toObjectErr(err, bucket, object)
 	}
-
-	onlineDisks, metaArr = shuffleDisksAndPartsMetadataByIndex(onlineDisks, metaArr, fi.Erasure.Distribution)
-
 	if fi.Deleted {
 		if opts.VersionID == "" {
 			return toObjectErr(errFileNotFound, bucket, object)
 		}
 		return toObjectErr(errMethodNotAllowed, bucket, object)
 	}
 
-	for i, fi := range metaArr {
-		if errs[i] != nil {
-			// Avoid disks where loading metadata fail
-			continue
-		}
-
-		// clean fi.Meta of tag key, before updating the new tags
-		delete(fi.Metadata, xhttp.AmzObjectTagging)
-		// Don't update for empty tags
-		if tags != "" {
-			fi.Metadata[xhttp.AmzObjectTagging] = tags
-		}
-		for k, v := range opts.UserDefined {
-			fi.Metadata[k] = v
+	onlineDisks, metaArr = shuffleDisksAndPartsMetadataByIndex(onlineDisks, metaArr, fi.Erasure.Distribution)
+	for i, metaFi := range metaArr {
+		if metaFi.IsValid() {
+			// clean fi.Meta of tag key, before updating the new tags
+			delete(metaFi.Metadata, xhttp.AmzObjectTagging)
```

Comment:
This is from the old code but doesn't seem to be needed, shouldn't we only update if tags is not empty ?

Reply:
There is no update all tags a replace only operations @vadmeste - delete is a no-op if key doesn't exist so its okay.